internal/subscription/endpoints/list_subscription.go

```diff
@@ -4,34 +4,96 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"time"
 
 	"github.com/go-kit/kit/endpoint"
 	kithttp "github.com/go-kit/kit/transport/http"
 	"github.com/google/uuid"
 	"github.com/sumelms/microservice-course/internal/subscription/domain"
 )
 
-type listSubscriptionRequest struct {
+type ListSubscriptionsRequest struct {
 	CourseUUID uuid.UUID `json:"course_uuid"`
 	UserUUID   uuid.UUID `json:"user_uuid"`
 }
 
-type listSubscriptionResponse struct {
-	Subscriptions []domain.Subscription `json:"subscriptions"`
+type SubscriptionCourseResponse struct {
+	UUID *uuid.UUID `db:"uuid" json:"uuid"`
+	Code *string    `db:"code" json:"code"`
```

Comment:
Is there any reason to use `*string` instead of `string` for `Code` and `Name`? Normally we use `*string` when we want something to be `nil` and then we use `omitempy`, but I don't see it here.

Reply:
https://github.com/sumelms/microservice-course/pull/28/commits/f25e64978fa73f9551d72ad49f3e2be53b12179b